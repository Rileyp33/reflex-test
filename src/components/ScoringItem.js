import React from 'react'
import { View, Text } from 'react-native'
import { ScaledSheet } from 'react-native-size-matters'

const ScoringItem = ({ label, containerColor, labelColor, dataColor, barColor, data }) => {
  return (
    <View style={[styles.textContainer, { backgroundColor: containerColor }]}>
      <View style={styles.label}>
        <Text style={[styles.gameDataText, { color: labelColor }]}>{label}</Text>
      </View>
      <View style={[styles.dataItem, { borderLeftColor: barColor }]}>
        <Text style={[styles.gameDataText, { color: dataColor }]}>{data}</Text>
      </View>
    </View>
  )
}

const styles = ScaledSheet.create({
  textContainer: {
    width: '49%',
    paddingVertical: '10@s',
    paddingHorizontal: '8@s',
    borderRadius: '4@s',
    flexDirection: 'row'
  },
  gameDataText: {
    fontSize: '14@s',
    fontWeight: 'bold',
    textAlign: 'justify'
  },
  dataItem: {
    width: '37%',
    borderLeftColor: 'white',
    borderLeftWidth: 1,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  label: {
    width: '63%%'
  }
})

export default ScoringItem