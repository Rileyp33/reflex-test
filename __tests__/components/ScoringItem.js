import React from 'react'
import renderer from 'react-test-renderer'
import ScoringItem from '../../src/components/ScoringItem'

test('Scoreboard item renders correctly', () => {
  const tree = renderer.create(
    <ScoringItem
      label={'Level'}
      data={1}
      containerColor={'#009FB7'}
      labelColor='white'
      dataColor='white'
      barColor='white'
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
