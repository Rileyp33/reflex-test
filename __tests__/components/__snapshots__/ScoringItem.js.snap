// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard item renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 8.571428571428571,
        "flexDirection": "row",
        "paddingHorizontal": 17.142857142857142,
        "paddingVertical": 21.428571428571427,
        "width": "49%",
      },
      Object {
        "backgroundColor": "#009FB7",
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "63%%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "justify",
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Level
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "white",
          "borderLeftWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "37%",
        },
        Object {
          "borderLeftColor": "white",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "justify",
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;
